scripts/leapp_upgrade.py

```diff
@@ -0,0 +1,382 @@
+import json
+import os
+import subprocess
+
+
+JSON_REPORT_PATH = "/var/log/leapp/leapp-report.json"
+TXT_REPORT_PATH = "/var/log/leapp/leapp-report.txt"
+REBOOT_GUIDANCE_MESSAGE = "A reboot is required to continue. Please reboot your system."
+
+# Based on https://github.com/oamg/leapp/blob/master/report-schema-v110.json#L211
+STATUS_CODE = {
+    "high": 3,
+    "medium": 2,
+    "low": 1,
+    "info": 0,
+}
+STATUS_CODE_NAME_MAP = {
+    "high": "ERROR",
+    "medium": "WARNING",
+    "low": "WARNING",
+    "info": "INFO",
+}
+
+
+# Both classes taken from:
+# https://github.com/oamg/convert2rhel-worker-scripts/blob/main/scripts/preconversion_assessment_script.py
+class ProcessError(Exception):
+    """Custom exception to report errors during setup and run of leapp"""
+
+    def __init__(self, message):
+        super(ProcessError, self).__init__(message)
+        self.message = message
+
+
+class OutputCollector(object):
+    """Wrapper class for script expected stdout"""
+
+    def __init__(self, status="", message="", report="", entries=None):
```

Comment:
We will need to update this the changes proposed by @andywaltlova here: https://github.com/oamg/convert2rhel-worker-scripts/commit/cff48732b5814f1267abe24b32f71f2463d76c7e

Reply:
Did this also now in https://github.com/oamg/leapp-insights-tasks/pull/4 so check it out